src/main/scala/nl.knaw.dans.easy.curationwork/Assign.scala

```diff
@@ -0,0 +1,76 @@
+/**
+ * Copyright (C) 2017 DANS - Data Archiving and Networked Services (info@dans.knaw.nl)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package nl.knaw.dans.easy.curationwork
+
+import java.nio.file.{Files, Path, Paths}
+
+import nl.knaw.dans.lib.logging.DebugEnhancedLogging
+import org.apache.commons.configuration.PropertiesConfiguration
+import org.apache.commons.csv.CSVFormat
+import org.apache.commons.io.FileUtils
+import resource.managed
+
+import scala.annotation.tailrec
+import scala.collection.JavaConverters._
+import scala.io.StdIn
+import scala.language.postfixOps
+import scala.sys.process._
+import scala.util.Try
+import scala.xml.XML
+
+
+class Assign(configuration: Configuration) extends DebugEnhancedLogging {
+
+  private def getCurationDirectory(datamanager: Option[DatamanagerId]): Path = {
+    datamanager.map(getManagerCurationDir).getOrElse(commonCurationDir)
+  }
+
+  private def getManagerCurationDir(datamanager: DatamanagerId): Path = {
+    Paths.get(managerCurationDirString.replace("$unix-user", datamanager))
+  }
+
+  private def directoryExists(dir: Path, uuid: Option[BagId] = None): Boolean = {
+    Files.exists(dir.resolve(uuid.getOrElse("")))
+  }
+
+  private def setProperties(depositProperties: PropertiesConfiguration, datamanager: DatamanagerId): Unit = {
+    val datamanagerProperties = configuration.datamanagers.getString(datamanager).split(" ")
```

Comment:
1. a `split` can also take a second parameter, which says in how many pieces it needs to split. Since you know this is 2, you can use that number here as second argument
2. Then you can pattern match on the output value. Rather than `val datamanagerProperties = ...` you can say `val Array(userId, email) = ...`. You can then use `userId` and `email` below instead of `datamanagerProperties(0)` and `datamanagerProperties(1)`. That's way more readable and better understandable as to what these values mean.

Reply:
A good point.  Now that I changed the way datamanager properties are stored in the properties file, this change is not anymore needed.